src/test/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMockTest.java

```diff
@@ -553,4 +553,211 @@ void runTask_AsyncConsumer() throws Exception
 		assertTrue(countDownLatch.await(2, TimeUnit.SECONDS));
 	}
 
+	@Test
+	void registerOneTasksAsynchronously()
+	{
+		final Thread mainThread = Thread.currentThread();
+
+		AtomicBoolean executed = new AtomicBoolean();
+		AtomicBoolean completed = new AtomicBoolean();
+		AtomicBoolean notPrimaryThread = new AtomicBoolean();
+		Thread thread = new Thread(() ->
+		{
+			try
+			{
+				scheduler.runTaskLater(null, bukkitTask ->
+				{
+					if (mainThread != Thread.currentThread())
+					{
+						notPrimaryThread.set(true);
+					}
+					executed.set(true);
+				}, 1);
+			}
+			finally
+			{
+				completed.set(true);
+			}
+		});
+		thread.start();
+
+		// Wait for thread to register the task
+		while (!completed.get())
+		{
+			scheduler.performOneTick();
+			Thread.yield();
+		}
+
+		while (!executed.get())
+		{
+			scheduler.performOneTick();
```

Comment:
Should we also add a Thread.yield() here?

Reply:
No, because the other thread has already terminated when this cycle executes.